### PR TITLE
dvmdostem update wrappers to increase output flexibility

### PR DESCRIPTION
**Pull request by tobeycarman**
## Description

Refactors the way that output specification was handled in the `dvmdostem` wrapper code. More control is now available to the user and the design should allow for the easy addition of new variables. In addition, there is a mechanism for defining relationships between `dvmdostem` variables and PEcAn variables (i.e. one PEcAn output is the sum of several different dvmdostem outputs).
* User control of output variables via `<dvmdostem_pecan_outputs>` tag in pecan.xml
* Tag accepts space separated string of PEcAn output variable names
* Wrapper code will parse user supplied string and will generate a dvmdostem output_spec.csv file for this run.
* Mapping between PEcAn and dvmdostem variables (names, units, depends etc) is maintained in an `R` list that can easily be added to.
* Mapping supports "depends_on" key which allows one PEcAn output to be assembled from several dvmdostem outputs. Currently only `sum` is implemented, but different relationships could be added.

## Motivation and Context

Previously the user was not able to easily control which output variables the `dvmdostem` model would produce and subsequently translate into PEcAn format.

## Review Time Estimate

- [ ] Immediately
- [x] Within one week
- [ ] When possible
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) 
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the CHANGELOG.md.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: models/dvmdostem/DESCRIPTION [hunk 1]
```diff
@@ -4,7 +4,7 @@ Title: PEcAn package for integration of the dvmdostem model
 Version: 1.7.0
 Date: 2018-12-03
 Authors@R: c(person("Shawn", "Serbin"),
-             person("Tobey", "Carmen"),
+             person("Tobey", "Carman"),
              person("James", "Simkins"),
              person("Chris", "Black"),
              person("Istem", "Fer"),
```
Comment (serbinsh) [inline models/dvmdostem/DESCRIPTION]: @tobeycarman confirm additional authors

File: models/dvmdostem/R/model2netcdf.dvmdostem.R [hunk 2]
```diff
@@ -6,14 +6,112 @@
...
+
 ##-------------------------------------------------------------------------------------------------#
 ##' @name model2netcdf.dvmdostem
 ##' @title Code to convert dvmdostem netcdf output into into CF standard
 ##' 
 ##' @param outdir Location of dvmdostem model output
 ##' @param runstart ??
 ##' @param runend ??
-##' 
+##' @param pecan_requested_vars a space separated string with names of the PEcAn variables to output.
```
Comment (serbinsh) [inline models/dvmdostem/R/model2netcdf.dvmdostem.R]: can you provide an example of you what want here?  I feel like space separated is a bad way to go and better to have a comma list unless there is a good reason?

Comment (tobeycarman) [reply models/dvmdostem/R/model2netcdf.dvmdostem.R]: @serbinsh, not sure why I chose space delimited here, but you are right probably a bad idea. I will change to comma.

Comment (tobeycarman) [reply models/dvmdostem/R/model2netcdf.dvmdostem.R]: @serbinsh, ok, just pushed a few more changes that should change to a comma separated implementation.

File: models/dvmdostem/R/model2netcdf.dvmdostem.R [hunk 3]
```diff
@@ -13,7 +13,7 @@ library(lubridate)
 ##' @title Write data into PEcAn shaped output file.
 ##' @param y_starts a list of years, i.e.: 1901, 1902, 1903, etc.
 ##' @param outdir a path to the location where were we will look for dvmdostem outputs and write PEcAn outputs.
-##' @param pecan_requested_vars space separated string listing the variables to process (PEcAn names).
+##' @param pecan_requested_vars comma separated string listing the variables to process (PEcAn names).
```
Comment (serbinsh) [inline models/dvmdostem/R/model2netcdf.dvmdostem.R]: @tobeycarman did you rebuild the docs after making the changes to the documentation?

Comment (tobeycarman) [reply models/dvmdostem/R/model2netcdf.dvmdostem.R]: I had not done that, but I think I finally got everything comitted and I think the build is passing now.

**Pull request merged**